### PR TITLE
AuthSig improvements

### PR DESCRIPTION
**Pull request by James-Mart**
This PR is mostly refactoring existing functionality:
1. Refactors the SPKI library that was previously coupled to AuthSig/RAuthSig, it is now its own library in the same namespace that can be independently linked
2. Minor improvement to the tester, starting to remove its dependency on the `psibase::PublicKey` type (secp256k1 keys), which is being deprecated.
3. Wraps `AuthSig` and `AuthK1` in their own namespaces, to better isolate functionality that is specific to each auth library.

And there are a few additions also related to auth:
1. Adds some functionality to the AuthSig plugin (The ability to set the key of the caller, and the ability to create and return a new keypair in pem format)
2. Adds a new library (also in AuthSig namespace) for parsing PEM to PrivateKeyInfo. Basically the private key equivalent to the SPKI library, although there's no packing/unpacking logic because it's intended to be used for signing in tests, not as objects that can packed and sent to the node.

### CODE REVIEW

File: services/system/AuthSig/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,19 @@
 function(add suffix)
     add_system_service("${suffix}" AuthSig src/AuthSig.cpp)
     add_system_service("${suffix}" RAuthSig src/RAuthSig.cpp)
+    
+    add_library(Spki${suffix} src/Spki.cpp)
```
Comment (swatanabe) [inline services/system/AuthSig/CMakeLists.txt:5]: The libraries also need to be installed and there should be a way to refer to them from `psidk`.

File: libraries/psibase/tester/include/psibase/DefaultTestChain.hpp [hunk 2]
```diff
@@ -35,11 +35,19 @@ namespace psibase
          }
       }
 
+      template <typename AuthService, typename AccountsService>
```
[the remark this answers is not in the document]
Comment (James-Mart) [reply libraries/psibase/tester/include/psibase/DefaultTestChain.hpp]: Not really. I've removed this and just included the Accounts service directly.

File: services/system/AuthSig/src/Spki.cpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+         }
+      }
+
+      std::string to_string(const SubjectPublicKeyInfo& key)
+      {
+         return Botan::PEM_Code::encode(reinterpret_cast<const std::uint8_t*>(key.data.data()),
+                                        key.data.size(), "PUBLIC KEY");
+      }
+
+      std::vector<unsigned char>& clio_unwrap_packable(SubjectPublicKeyInfo& obj)
```
[the remark this answers is not in the document]
Comment (James-Mart) [reply services/system/AuthSig/src/Spki.cpp]: Inlined

File: services/system/AuthSig/CMakeLists.txt (hunk 1 above)
Comment (James-Mart) [reply services/system/AuthSig/CMakeLists.txt:5]: Okay, I believe this is addressed

**Review by swatanabe — APPROVED** [review]

**Pull request merged**